### PR TITLE
Course query endpoint

### PR DESCRIPTION
**Pull request by xenshreyas**
Addresses https://github.com/ubclaunchpad/forum/issues/86

### CODE REVIEW

File: backend/routers/routes/courses.py [hunk 1]
```diff
@@ -1,9 +1,98 @@
...
+    if not user_has_access_to_course(user_id, course_id):
+        raise HTTPException(status_code=403, detail="Access denied to this course")
+
+    # Initialize DocumentQueryEngine
+    try:
+        query_engine = DocumentQueryEngine(max_chunks=query_request.max_chunks)
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Error initializing query engine: {e}"
+        )
```
Comment (armintalaie) [inline backend/routers/routes/courses.py]: I think this should be initialized outside the post endpoint. it doesn't need to be init'd for every request.

File: backend/routers/routes/courses.py [hunk 2]
```diff
@@ -1,9 +1,98 @@
...
+    course_id: str, query_request: CourseQueryRequest, request: Request
+):
+    """
+    Endpoint for querying course content using RAG.
+    """
+    user_id = request.headers.get("X-User-ID")
+
+    if not user_has_access_to_course(user_id, course_id):
+        raise HTTPException(status_code=403, detail="Access denied to this course")
+
```
Comment (armintalaie) [inline backend/routers/routes/courses.py:75]: good but let's comment it out for now to test

Comment (xenshreyas) [reply backend/routers/routes/courses.py:75]: the implementation for this function currently just returns `True` so should be the same :D

File: backend/routers/routes/courses.py (hunk 1 above)
Comment (xenshreyas) [reply backend/routers/routes/courses.py]: we actually need to get the max chunks from the query itself, but I agree, if possible we should initialize outside of the endpoint

**Review by armintalaie — APPROVED** [review]

**Pull request merged**